pkg/detectlicense/licenses.go

```diff
@@ -82,14 +82,15 @@ loop:
 		name := filepath.Base(filename)
 		// See ./vendor.go:metaPrefixes
 		switch {
-		case strings.HasPrefix(name, "AUTHORS") ||
+		case strings.HasPrefix(name, "AUTHORS"),
 			strings.HasPrefix(name, "CONTRIBUTORS"):
 			// Ignore this file; it does not identify a license.
-		case strings.HasPrefix(name, "COPYLEFT") ||
-			strings.HasPrefix(name, "COPYING") ||
-			strings.HasPrefix(name, "COPYRIGHT") ||
-			strings.HasPrefix(name, "LEGAL") ||
-			strings.HasPrefix(name, "LICENSE"):
+		case strings.HasPrefix(name, "COPYLEFT"),
+			strings.HasPrefix(name, "COPYING"),
+			strings.HasPrefix(name, "COPYRIGHT"),
+			strings.HasPrefix(name, "LEGAL"),
+			strings.HasPrefix(name, "LICENSE"),
+			strings.EqualFold(name, "license.md"): // Explicitly named license file (case-insensitive).
```

Comment:
I guess the `./` part of that "See ./vendor.go:metaPrefixes" is outdated, but what it's referring to is https://github.com/datawire/go-mkopensource/blob/e7c9b5e549afcd6a1c3e9554eaec27be7dacc7dd/vendor.go#L80-L100

The general theme of `vendor.go` is to mimic the internals of the `go mod vendor` command; the theme here is to mimic that too.  For all of those existing `HasPrefix`s, the file would be copied in to `vendor/` via `go mod vendor`.  We're not doing it right now, but one of the capabilities of go-mkopensource that is important is that it can generate a tarball with all of the licenses in it; not just a listing of the licenses.

So adding it here only half addresses the problem.  I'd prefer to keep the main logic "vanilla" `go mod vendor`, and add a special case for whichever package contains a `license.md`.